hera_pspec/utils.py

```diff
@@ -116,7 +116,8 @@ def variance_from_auto_correlations(uvd, bl, spw_range, time_index):
     x_bl1 = uvd.get_data(bl1)[time_index, spw]
     x_bl2 = uvd.get_data(bl2)[time_index, spw]
     nsample_bl = uvd.get_nsamples(bl)[time_index, spw]
-    nsample_bl = np.where(nsample_bl>0, nsample_bl, np.median(uvd.nsample_array[:,:,spw,:]))
+    nsample_bl = np.where(nsample_bl>0, nsample_bl, np.median(uvd.nsample_array[:, spw, :]))
+    df = np.array(uvd.channel_width)[spw]
```

Comment:
do you really want this to be an array? if not, `np.median(np.diff(uvd.freq_array))` might be what you want

Reply:
I chose to follow the new `UVData` syntax inside the `UVPSpec` objects and make `channel_width` an array (although, most of the time, it will be an array of identical values). I added broadcasting to the next equation to make array shapes clearer.